### ISSUE TITLE
Coords change with timewarp

### ISSUE THREAD

**Issue by fronbow** [Ezriilc/HyperEdit#34]
As per @wile1411

> For the Loading of Co-ordinates, I don't think the co-ordinates used are fixed to the planet surface. If I load up a set of Co-ordinates over land, it works fine. If I time warp a day or so, the same co-ordinates will be somewhere else on the planet (possibly over water) as the planet rotates away from desired coordinate location.
> 
> I suspect there is some extra transform maths needed to get the desired location to be aligned with the planets surface rotation.

**Comment by wile1411** [comment]
Steps to reproduce:

 - Load any old craft onto the launch pad
 - open HpterEdit Lander Screen
 - click load and select one of the preset coordinates with ALT set to 5000 to avoid the other issue with trying to spawn under the water (see comment in issue #29)
 - Click LAND and ship will spawn in 'a location'
 - Once confirmed that this location is not over water - Fiddle with the ALT so you can actually land in a reasonable time and actually land.
 - Time warp to the following morning (Leave HyperEdit Landing window open)
 - Click LAND again (coordinates unchanged) and you end up in a different location compared to where you just landed.
 
Hopefully that helps! :)

**Comment by fronbow** [comment]
Yes this helped. But I have a feeling you're using the latest stable release and not the latest beta as I'm getting expected results.
I am curious why you would want to land at 5000m though?!

**Comment by wile1411** [comment]
I was using the Beta2 version from the KerbalTek site. I know I 100% deleted the old version and renamed the new file in it's place. 
I was setting to 5000 to avoiding the below surface error to find a location that was above Land. I had a few instances where Hyperedit didn't catch my spawning in a surface and blew up the ship I was testing with. The 5000 was mainly me trying to avoid having to reload a new ship each time it blew up - I'm incredibly crap at remembering to use F5/F9, so I always end up having to revert and try again when this happens.

After I found a location above land, I'd reduce it to 20m as I had a little more faith that I'd not spawn under ground.

**Comment by wile1411** [comment]
Also can confirm it wasn't the most recent release as repeated click of LAND button at 20m above surface for the Beta2 seemed to give the same location each time. 

With the latest release it would change by a few 100/1000m (don't know exact) for each click. Any time I did use LAND with the current version and was near a hill, I'd get nervous know the spawning site might move me into said hill

**Comment by fronbow** [comment]
The coords do change by about 50m after every repeated Land. This looks like a safety issue for when ppl are landing lots of vessels near each other.

**Comment by wile1411** [comment]
Hmmm.. that sounds less than what it used to be when I previously tested on Minmus, so I guess an improvement? I remember the last time I used it that the hills around the flats would change a good amount after each load, but that might be due to Minmus being smaller radius compared to Kerbin.

I bit if you did the repeated landings on something like Gilly it'd be a lot more than a 50m difference. o.O

**Comment by fronbow** [comment]
Well that's why we need testers ;)
I put the dll version number (1.5.5.1) in the other thread - this is the latest beta.
From what I remember (got too many windows open spread across 2 screens lol) the 50m distance is calculated relative to the planet so it should be the same.
A future idea could be to reduce this "magic" distance relative to the planet/moon.

**Comment by wile1411** [comment]
sorry - didn't mean to get you to test, was only thinking what I'll do to test out aloud. I'm away from the KSP machine at work and can't test at the moment. Will give it ago tonight (add ~13hs from now and I'll post an update on testing here)

**Comment by fronbow** [comment]
Blimey that's a long work-day. I'm in the UK, so I'll probably be afk at that time.

**Comment by wile1411** [comment]
Nah - normal day and just started, but finish at 5. Almost 9am here (Canberra) and can't get to the KSP until son finally goes to sleep after 9pm-ish

**Comment by fronbow** [comment]
Ahh, well, hope your work day is a good one :)

**Comment by Ezriilc** [comment]
I'm wondering if this issue is related to:  https://github.com/Ezriilc/HyperEdit/issues/14
